rclcpp/src/rclcpp/executors/multi_threaded_executor.cpp

```diff
@@ -70,6 +70,42 @@ MultiThreadedExecutor::get_number_of_threads()
   return number_of_threads_;
 }
 
+// Iterate through all the callback groups of
+// all nodes in the executor and verify
+// there is a non-empty callback group
+// that is "Mutually Exclusive". If there is,
+// then set trigger guard condition to true
+// so that it avoids a thread getting stuck
+// in rcl_wait.
+bool
+MultiThreadedExecutor::determine_wake_after_execute()
```

Comment:
Does this need to return the value? It seems like `get_wake_after_execute_flag()` is used in most places instead.

If not, then maybe this should be called `void evaluate_need_to_wake_after_execute()`?